trino/trino.go

```diff
@@ -1943,3 +2017,19 @@ func (s *NullSlice3Map) Scan(value interface{}) error {
 	s.Valid = true
 	return nil
 }
+
+type QueryProgressCallbackResponse struct {
+	ID             string
+	QueryStats     stmtStats
+	LastQueryState string
+}
+
+type queryProgressCallbackPeriod struct {
+	Period           time.Duration
+	LastCallbackTime time.Time
+}
+
+type ProgressUpdater interface {
+	// Update : Called on progress update, before and after query execution finished.
```

Comment:
Nit: the comment should start with the function name but it should be a valid English sentence, so use something like:
```suggestion
	// Update the query progress, immediately when the query starts, when receiving data, and once when the query is finished.
```